Porting TPC Trending to QC Framework as extended Trending

mlesch opened the pull request.

knopers8 commented on Framework/src/TrendingTaskExtended.cxx:
```diff
@@ -0,0 +1,607 @@
+  }
+
+  mTrend->Fill();
+} // void TrendingTaskExtended::trendValues(uint64_t timestamp, repository::DatabaseInterface& qcdb)
```
leftover?

mlesch replied:
Changed as requested

knopers8 commented on Framework/include/QualityControl/TrendingTaskExtended.h:
```diff
@@ -0,0 +1,97 @@
+/// and input/output canvas can be dealt with alongside normal histograms.
+///
+
+class TrendingTaskExtended : public PostProcessingInterface
```
How about `SliceTrendingTask`?

mlesch replied:
Changed as requested

knopers8 reviewed (changes requested):
Thanks a lot for this @mlesch! I think this is mostly good as it is, I have mostly some editorial comments about better naming. I have not tried yet to run it on my setup, but I am compiling it now and will let you know if I run into problems when executing it.

I would also like to ask you to add some documentation in this chapter: https://github.com/AliceO2Group/QualityControl/blob/master/doc/PostProcessing.md#convenience-classes
It would be nice if you could at least summarize what does the class do and how to configure it.

knopers8 commented:
Hi @mlesch , I played with the new task now and I am not sure if I get the results I should expect.
First of all, `ExtendedTrending_meanX_of_histogram` shows always one value for each point, even though I am running the example QC Task in parallel, so I would expect it to change. However, if I execute the ExtendedTrendingTask again, I see it picks different values. Am I doing something wrong? All the other plots always get the same values as well.

On `ExtendedTrending_meanX_of_histogram` I had a problem to see the data points actually, but this was fixed after I set `graphYRange` to none. Perhaps you could tune these parameters better in the example or leave them empty as well?

On `ExtendedTrending_meanY_of_histogram_slices` there are `slices` on the x axis, which I find a bit misleading, we still represent some parameter with a unit there, not an identifier of a slide. Perhaps the label could be more precise?

On `ExtendedTrending_meanY_of_histogram_timeMultigraph` I also had to put empty  `graphYRange`to see something, perhaps it could be left empty in the example.

I am sorry if I am nitpicking, but I believe that a clear example will be crucial to advertize the class well to other the detector teams.

mlesch commented on Framework/include/QualityControl/SliceInfoTrending.h:
```diff
@@ -0,0 +1,113 @@
+  std::string title = "";
+
+  /// \brief Return the struct member/float corresponding to the argument.
+  double RetrieveValue(const std::string& varType) const
```
Changed as requested

mlesch commented on Framework/include/QualityControl/TrendingTaskExtended.h:
```diff
@@ -0,0 +1,97 @@
+#include "QualityControl/SliceInfoTrending.h"
+#include "QualityControl/TrendingTaskExtendedConfig.h"
+
+#include <memory>
```
Changed as requested

mlesch commented on Framework/include/QualityControl/ReductorExtended.h:
```diff
@@ -0,0 +1,68 @@
+/// A extended reductor class from which each reductor used for the extended trending.
+///
+
+class ReductorExtended
```
Changed as requested

mlesch commented on Modules/Common/include/Common/QualityReductorExtended.h:
```diff
@@ -0,0 +1,47 @@
+/// a struct of 'SliceInfoQuality'
+///
+
+class QualityReductorExtended : public quality_control::postprocessing::ReductorExtended
```
Removed as requested as it does not bring anything new compared to the framework trending

mlesch commented:
Hi @knopers8 , 
I implemented the requested changes and fixed a bug. Now the trending changes over time when running the QC Task in parallel. I also tried to precise the labels of the plots in the json file a bit better, but of course I can change them if the current version is not clear enough. I will now start working the documentation about the class.
